multiversx_sdk/core/controllers/smart_contract_controller.py

```diff
@@ -0,0 +1,226 @@
+        transaction = transaction_awaiter.await_completed(tx_hash)
+        return self.parse_deploy(transaction)
+
+    def create_transaction_for_upgrade(self,
+                                       sender: IAccount,
+                                       nonce: int,
+                                       contract: IAddress,
+                                       bytecode: Union[Path, bytes],
+                                       gas_limit: int,
+                                       arguments: Sequence[Any] = [],
+                                       native_transfer_amount: int = 0,
+                                       is_upgradeable: bool = True,
+                                       is_readable: bool = True,
+                                       is_payable: bool = False,
+                                       is_payable_by_sc: bool = True) -> Transaction:
+        self._ensure_factory_is_initialized()
+
+        transaction = self.factory.create_transaction_for_upgrade(  # type: ignore
+            sender=sender.address,
+            contract=contract,
+            bytecode=bytecode,
+            gas_limit=gas_limit,
+            arguments=arguments,
+            native_transfer_amount=native_transfer_amount,
+            is_upgradeable=is_upgradeable,
+            is_readable=is_readable,
+            is_payable=is_payable,
+            is_payable_by_sc=is_payable_by_sc
+        )
+
+        transaction.nonce = nonce
+        transaction.signature = sender.sign(self.tx_computer.compute_bytes_for_signing(transaction))
+
+        return transaction
+
+    def create_transaction_for_execute(self,
+                                       sender: IAccount,
+                                       nonce: int,
+                                       contract: IAddress,
+                                       gas_limit: int,
+                                       function: str,
+                                       arguments: Sequence[Any] = [],
+                                       native_transfer_amount: int = 0,
+                                       token_transfers: Sequence[TokenTransfer] = []) -> Transaction:
+        self._ensure_factory_is_initialized()
+
+        transaction = self.factory.create_transaction_for_execute(  # type: ignore
+            sender=sender.address,
+            contract=contract,
+            gas_limit=gas_limit,
+            function=function,
+            arguments=arguments,
+            native_transfer_amount=native_transfer_amount,
+            token_transfers=token_transfers
+        )
+
+        transaction.nonce = nonce
+        transaction.signature = sender.sign(self.tx_computer.compute_bytes_for_signing(transaction))
+
+        return transaction
+
+    def parse_execute(self, transaction_on_network: TransactionOnNetwork, function: Optional[str] = None):
+        pass
+
+    def await_completed_execute(self, tx_hash: str):
+        pass
```

Comment:
Not implemented, perhaps throw an exception? Here & above?

Reply:
done